roles/servicetelemetry/templates/manifest_smartgateway_events.j2

```diff
@@ -25,9 +25,9 @@ spec:
     name: elasticsearch
   bridge:
     amqpUrl: amqp://{{ meta.name }}-interconnect.{{ meta.namespace }}.svc.cluster.local:5673/{{ this_collector.subscription_address }}
-    ringBufferSize: {{ this_collector.bridge.ring_buffer_size }}
-    ringBufferCount: {{ this_collector.bridge.ring_buffer_count }}
-    verbose: {{ this_collector.bridge.verbose }}
+    ringBufferSize: {{ this_collector.bridge.ring_buffer_size | default(16384) }}
+    ringBufferCount: {{ this_collector.bridge.ring_buffer_count | default(15000) }}
+    verbose: {{ this_collector.bridge.verbose | default(false) }}
```

Comment:
Normally we wouldn't use inline defaults in the templates, but for how this template gets utilized I think it is appropriate.

We do set defaults for the values of the default defined cloud in `defaults/main.yml` but since that entire stanza can be overridden, and we don't have control on what is specifically passed in, we need to provide a set of defaults for the parameters that may be missing.

Especially since this is an expansion on the parameters that would normally be provided prior to this release, we want to make sure defaults are present for the existing CR definitions that would not have had the ringBuffer params specified.

Will be testing to make sure defining an events cloud without the parameters won't cause another issue where Ansible complains that `this_collector.bridge.ring_buffer_size` is not defined (in which case we'll need to add some other logic that checks for the parameter being defined, and if not, then use the defaults). Hoping this is enough though to cover that use case.

Reply:
I think this will work and the argument to duplicate the defaults in the template in this case makes sense.

It does make me wonder if there are other parameters that could benefit from in-template defaults because of the ability to override entire stanzas of parameters.

Reply:
It _should_ only be required in the situation where you have lists of dictionaries where the defaults won't push through.

So in this case, that is only for the `clouds` parameter which is a list of dictionaries for each of the SGs. In the case of everything else (https://github.com/infrawatch/service-telemetry-operator/blob/master/roles/servicetelemetry/defaults/main.yml#L9-L82) those should not need template-based defaults because of the way we set up in https://github.com/infrawatch/service-telemetry-operator/blob/master/roles/servicetelemetry/vars/main.yml

We even have the note at https://github.com/infrawatch/service-telemetry-operator/blob/master/roles/servicetelemetry/defaults/main.yml#L84-L85 which implies the entire structure is overridden when defined.

So I think we mostly just need to be careful for instances where `clouds.*` is referenced.